### PR TITLE
test: Testing support for float32

### LINKED ISSUE #53: Proposal: support float32
`float32` is not officially supported in Thrift, but it's a valid type in Hessian2 Codec.

To support float32, we have to extend thriftgo (with `-thrift float32` flag) for generating float32 structs (only for struct fields, no need to implement the encoding/decoding of float32 for thrift).

After that, we can add float32 support in codec-dubbo.

### PR DESCRIPTION
**Pull request by aaravarr**
#### What type of PR is this?

test

#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [ ] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io)

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en:
zh(optional): 

#### (Optional) Which issue(s) this PR fixes:

Fixes #53

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,36 +20,36 @@
...
-| map\<bool, string> | map[bool]string  |     map     | Map\<Boolean, String>  |    HashMap\<Boolean, String>    |
+|     thrift 类型      |    golang 类型     | hessian2 类型 |       默认 java 类型       |                可拓展 java 类型                 |
+|:------------------:|:----------------:|:-----------:|:----------------------:|:------------------------------------------:|
+|        bool        |       bool       |   boolean   |   java.lang.Boolean    |                  boolean                   |
+|        byte        |       int8       |     int     |     java.lang.Byte     |                    byte                    |
+|        i16         |      int16       |     int     |    java.lang.Short     |                   short                    |
+|        i32         |      int32       |     int     |   java.lang.Integer    |                    int                     |
+|        i64         |      int64       |    long     |     java.lang.Long     |                    long                    |
+|       double       |     float64      |   double    |    java.lang.Double    |    double <br> float / java.lang.Float     |
```
Comment (felix021) [inline README.md:30]: why `java.lang.Float`?

File: README.md [hunk 2]
```diff
@@ -20,36 +20,36 @@
...
+| map\<bool, string> | map[bool]string  |     map     | Map\<Boolean, String>  |         HashMap\<Boolean, String>          |
 
 **重要提示**：
 
 1. 映射表中的 map 类型并没有被完全列举，当前仅包含经过测试的用例。
 
 2. 不支持在 map 类型中使用包含和 **binary** 类型的键值。
 
-3. dubbo-java 不支持对包含 **i8**、**i16** 键值的 map 类型解码，建议避开 dubbo-java 不兼容的用法，可以在定义接口的响应字段时使用 **struct** 来包裹 map。
+3. 由于 **float32** 在 thrift 中不是有效的类型，DubboCodec 将 **float**(java) 映射到了 **float64**(go)，可以在 idl 中使用方法注解指定 **double** 映射为 **float**。
```
Comment (felix021) [inline README.md]: Please add a link to an example.

File: tests/crosstest/dubbo2kitex/base_type_test.go [hunk 3]
```diff
@@ -105,6 +105,12 @@ func TestEchoInt64(t *testing.T) {
 	assertEcho(t, err, req, resp)
 }
 
+func TestEchoFloat(t *testing.T) {
+	var req float64 = 12.3456
```
Comment (felix021) [inline tests/crosstest/dubbo2kitex/base_type_test.go]: Avoid using such values which can not be stored precisely in float/double. `1.0` will be a better choice.

Same below.

File: tests/dubbo-java/src/main/java/org/apache/dubbo/tests/api/UserProvider.java [hunk 4]
```diff
@@ -1,137 +1,153 @@
-/*
```
Comment (felix021) [inline tests/dubbo-java/src/main/java/org/apache/dubbo/tests/api/UserProvider.java]: Why are these files modified completely?

Comment (felix021) [reply tests/dubbo-java/src/main/java/org/apache/dubbo/tests/api/UserProvider.java]: At least the license header looks the same as before.

File: README.md (hunk 1 above)
Comment (aaravarr) [reply README.md:30]: `float` 和 `java.lang.Float` 都可以

Comment (felix021) [inline README.md:30]: Don't forget to update the English version accordingly.

**Review by felix021 — APPROVED** [review]
